miniapps/autodiff/fdual.hpp

```diff
@@ -0,0 +1,548 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef FDUAL_H
+#define FDUAL_H
+
+#include <cmath>
+#include <type_traits>
+
+namespace mfem
+{
+namespace ad
+{
+/** The FDual template class provides forward automatic differentiation
+   implementation based on dual numbers. The derivative of an arbitrary
```

Comment:
How about include an external link to reference material?
```html
<a href="https://en.wikipedia.org/wiki/Automatic_differentiation">automatic differentiation</a>
```

Reply:
Added to the documentation